Automattic/wp-calypso
# Update Akismet(site-less) purchase in me/purchases so that it can be viewed and managed.

This PR allows for Akismet (site-less) purchases to be viewed on the `me/purchases` page in Calypso and can click the purchase to view the purchase detail page at `/me/purchases/:site/:purchaseId`. 

Prior to this PR, clicking the Akismet purchase would do nothing, when it is supposed to take you to the purchase detail page, where you can view additional details as well as cancel or renew the subscription.
We also don't need to display the site name on the purchase when managing (siteless.akismet.com).

**Important Note:  The ability to _Cancel_ and _Renew_ the subscription purchase will need to be done in a follow-up PR's/Patches.

- See "**Before**" screenshot here:

![Markup 2023-03-17 at 15 18 02](https://user-images.githubusercontent.com/11078128/225999356-ec8045c5-34a1-4a4b-b72c-3fc6d4f50973.png)

- See "**After**" Screenshot(s) here:

![Screenshot on 2023-03-21 at 12-56-44](https://user-images.githubusercontent.com/11078128/226684878-6032345d-c5e5-4e7d-9d12-cce7ca387857.png)

<hr/>

![Markup 2023-03-19 at 15 48 14](https://user-images.githubusercontent.com/11078128/226205208-246717a2-9b64-41a9-a7d3-f5e97bc7ea18.png)


Asana Task: 1201210512993648-as-1204136657007224/f

Additional notes:
Much of all main files & components in the `client/me/purchases/` directory are still built using React classes.  Ideally we should refactor all these to functional components, however I think it would be best to do this refactoring as a maintenance task in the future because it would be quite an undertaking as these files are all quite large and complex, and they should be refactored and tested without introducing additional changes.

## Testing Instructions



- Checkout this PR/branch and spin up Calypso.
    - Run `git fetch && git checkout add/akismet-me-purchases && yarn start`
- Go to `calypso.localhost:3000/checkout/akismet/ak_plus_yearly_1`. and purchase Askimet (siteless).  Once you complete the purchase you should be redirected to a thank you page `http://calypso.localhost:3000/checkout/akismet/thank-you/ak_plus_yearly_1` which will be blank. The page does not exist yet (coming soon).
- Navigate to `/me/purchases` and look for your "**Akismet Plus (10K requests/month)**" purchase. (see screenshots above).
- Click into the "**Akismet Plus (10K requests/month)**" purchase, and verify you are directed to the purchase detail page (`/me/purchases/siteless.akismet.com/{:purchaseId}`). Verify the "Renew now" and "Cancel subscription" buttons are showing (Although the ability to renew and cancel siteless Akismet purchases are not working yet.) 



## Pre-merge Checklist
